packages/invoices/src/OutboundShipment/DetailView/DetailView.tsx

```diff
@@ -227,6 +235,38 @@ export const OutboundShipmentDetailViewComponent: FC = () => {
             <span>Transport details coming soon..</span>
           </Box>
         </TabPanel>
+        <Box display="flex" alignItems="flex-end" height={40}>
+          <Box flex={1} display="flex" justifyContent="flex-end" gap={2}>
+            <ButtonWithIcon
+              Icon={<XCircleIcon />}
+              labelKey="button.cancel"
+              variant="outlined"
+              color="secondary"
+              onClick={() => navigate(-1)}
+            />
+            <ButtonWithIcon
+              Icon={<Download />}
+              labelKey="button.save"
+              variant="contained"
+              color="secondary"
+              onClick={() => {
+                success('Saved invoice! 🥳 ')();
+                save(draft);
+              }}
+            />
+            <ButtonWithIcon
+              Icon={<ArrowRightIcon />}
+              labelKey="button.save-and"
+              variant="contained"
+              color="secondary"
+              onClick={() => {
+                success('Saved invoice! 🥳 ')();
+                draft.update?.('status', 'finalised');
+                save(draft);
+              }}
+            />
+          </Box>
+        </Box>
```

Comment:
aaanndd what I was actually meaning to do. A few buttons in the UI and think it might be best to follow up in a new PR with some actual logic.. They are sending requests, just not nicely.